### PR TITLE
Fix corruption of Unicode characters on Windows systems

### PR DESCRIPTION
**Pull request by hakanai**
(derived from previous PR, includes the cure but not the prevention)

I had an issue where a corrupted copy of ClientHooks.java was found inside
a jar file:

...\build\tmp\expandedArchives\forge-1.16.3-34.1.0_mapped_snapshot_20201028-1.16.3-sources.jar_d036fe06dff3dbdda94c44ac227acd65\net\minecraftforge\fml\client\ClientHooks.java:183:
error: unmappable character (0x81) for encoding UTF-8
return description.endsWith(":NOFML??r") ? description.substring(0,
description.length() - 8)+"??r" : description;

Best practices for file I/O streams is always to specify the character
set. I think it's reasonable to just enforce UTF-8 in all situations, so I
fixed that.

### CODE REVIEW

**Comment by hakanai** [comment]
The branch with the build script changes is here: https://github.com/MinecraftForge/ForgeGradle/compare/FG_3.0...ephemeral-laboratories:encoding-bug-prevention?expand=1

I won't submit a PR at this point because first the project should decide for itself whether it wants checks like this as part of its build.

**Comment by covers1624** [comment]
Can you please Rebase your changes onto the `FG_4` branch.

**Comment by hakanai** [comment]
Not sure what was going on there. GitHub told me I'd added a workflow file, then when I tried to undo that it said I had deleted it, then I had to undo that.

File: src/common/java/net/minecraftforge/gradle/common/task/ArchiveChecksum.java
Comment (covers1624) [inline src/common/java/net/minecraftforge/gradle/common/task/ArchiveChecksum.java]: You have heaps of * import collapses throughout here, we don't collapse imports to *, can you un collapses all these please?

**Review by covers1624 — CHANGES_REQUESTED** [review]

File: src/common/java/net/minecraftforge/gradle/common/task/ArchiveChecksum.java
Comment (hakanai) [reply src/common/java/net/minecraftforge/gradle/common/task/ArchiveChecksum.java]: Existing imports in several other files contradict that claim, but in any case, done.

**Comment by SizableShrimp** [comment]
This needs retargetting and rebasing on FG 5 to be considered.

**Comment by hakanai** [comment]
I've decided to close the PR.

A few reasons:
- I found that I had some calls which weren't just fixing the encoding but other locale-related issues (i.e., I should have cut the PR down further anyway)
- New calls to the bad methods appeared and I am not that keen on fixing new bad calls every release only to find that the PR isn't accepted by the next major release anyway (before attempting to fix this again, some sort of static checks to detect the issue really need to be in place or people will keep reimplementing the same issue)
- I have stopped supporting Forge in my own mods, so this issue no longer directly affects me

**Pull request closed without merging**
